internal/api/schemas/v1.yaml

```diff
@@ -11,17 +11,168 @@ info:
     name: Mozilla Public License 2.0
     url: https://www.mozilla.org/en-US/MPL/2.0/
 paths:
+  "/namespaces/{namespace}/gateways":
+    get:
+      description: 'Returns all configured gateways in the given namespace.'
+      operationId: listNamespacedGateways
+      parameters:
+      - name: namespace
+        in: path
+        description: Namespace of gateways to fetch
+        required: true
+        schema:
+          type: string
+      responses:
+        '200':
+          description: gateway response
+          content:
+            application/json:
+              schema:
+                "$ref": "#/components/schemas/GatewayPage"
+        default:
+          description: unexpected error
+          content:
+            application/json:
+              schema:
+                "$ref": "#/components/schemas/Error"
   "/gateways":
     get:
       description: 'Returns all configured gateways.'
       operationId: listGateways
+      parameters:
+      - name: namespaces
+        in: query
+        description: The namespaces of gateways to list
+        schema:
+          type: string
+      responses:
+        '200':
+          description: gateway response
+          content:
+            application/json:
+              schema:
+                "$ref": "#/components/schemas/GatewayPage"
+        default:
+          description: unexpected error
+          content:
+            application/json:
+              schema:
+                "$ref": "#/components/schemas/Error"
+    post:
+      description: 'Creates a new gateway.'
```

Comment:
So, we have options here, we can either leave namespaces to _only_ be specified via the path, or (which I kind of prefer), have it be a part of the request body parameter (i.e. `Gateway.namespace`) since it'll need to be associated with the Gateway/Routes either way. I know it makes it so there's not complete symmetry between the namespaced/non-namespaced routes, but I figured in this case it'd probably be alright, at least for now.

Reply:
I don't see a benefit of breaking from consistent patterns in this case. If every `Gateway` _must_ have a namespace, then I'd prefer we either
- always use the namespace in the path (and maybe alias just for the default namespace), or
- always use the namespace in the payload

Reply:
So, that's the thing that's a bit wonky about how Consul works under the hood -- technically yes, namespaces _are_ always required, but in OSS if you set a namespace to anything other than the magic default namespace, you get an error. Furthermore, if you don't specify a namespace you wind up with the default one getting implicitly added for you.

So, in OSS, despite them being required you _never_ deal with namespaces. What this means from our perspective just so we don't introduce concepts that don't exist for a bunch of our users is that you should be able to create a Gateway/Route/whatever without specifying a namespace. If, however, you _do_ specify a namespace, it becomes part of its identifier.

Why I have a preference towards making this the only different endpoint is that it seems a bit odd to me to specify all other Gateway parameters in the request body and then merge the path-based namespace onto the payload.